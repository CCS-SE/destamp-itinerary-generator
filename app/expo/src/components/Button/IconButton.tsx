import React from 'react';
import { TouchableOpacity, View } from 'react-native';

interface CustomButtonProps {
  onPress: () => void;
  icon: React.JSX.Element;
}

export default function IconButton({ onPress, icon }: CustomButtonProps) {
  return (
    <View className="items-center p-3">
      <TouchableOpacity
        accessibilityRole="button"
        activeOpacity={0.7}
        onPress={onPress}
        className="w-[120] items-center rounded-2xl bg-gray-200 p-2"
      >
        {icon}
      </TouchableOpacity>
    </View>
  );
}
