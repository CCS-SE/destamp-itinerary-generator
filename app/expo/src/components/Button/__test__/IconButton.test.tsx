import React from 'react';
import { AntDesign } from '@expo/vector-icons';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import IconButton from '../IconButton';

describe('Icon Button', () => {
  it('should render successfully', () => {
    const buttonCallbackFunction = jest.fn(() => null);
    const { getByRole } = render(
      <IconButton
        icon={<AntDesign name="heart" />}
        onPress={buttonCallbackFunction}
      />,
    );
    const iconBtnElement = getByRole('button');

    expect(iconBtnElement).toBeDefined();
  });

  it('should render correct icon', () => {
    const buttonCallbackFunction = jest.fn(() => null);
    render(
      <IconButton
        icon={<AntDesign name="heart" />}
        onPress={buttonCallbackFunction}
      />,
    );

    const heartIcon = renderer.create(<AntDesign />).toJSON();

    expect(heartIcon).toMatchSnapshot();
  });
});
