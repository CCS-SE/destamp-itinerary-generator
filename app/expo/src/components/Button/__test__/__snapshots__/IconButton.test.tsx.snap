// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button should render correct icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      {
        "fontFamily": "anticon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
/>
`;
